### PR TITLE
use utf8 encoding to serialize strings

### PR DESCRIPTION
**Pull request by peterjah**

### CODE REVIEW

**Review by Thykof — CHANGES_REQUESTED** [review]

**Review by Thykof — APPROVED** [review]

**Review by SebastienValla — COMMENTED** [review]
why do we want to go with UTF8 and not UTF16 ?

**Comment by peterjah** [comment]
> why do we want to go with UTF8 and not UTF16 ?

what do you mean by "to go with  UTF8". The proposal here is serializing /deserializing utf16 strings

File: packages/as-types/assembly/__tests__/serializer.spec.ts [hunk 1]
```diff
@@ -20,10 +20,15 @@ import {
 } from '../serialization';
 
 describe('Serialization tests', () => {
-  it('ser/deser string', () => {
+  it('ser/deser Utf-16 string', () => {
     const str = 'Hello world 🙂';
     expect(bytesToString(stringToBytes(str))).toBe(str);
   });
+  it('ser/deser Utf-8 string', () => {
```
Comment (gregLibert) [inline packages/as-types/assembly/__tests__/serializer.spec.ts:27]: It's rather an Ascii string ([a-zA-Z0-9] catalog to be precise). If you really want an UTF8 you should use some emoji stuff

File: packages/as-types/assembly/__tests__/serializer.spec.ts [hunk 2]
```diff
@@ -20,10 +20,15 @@ import {
 } from '../serialization';
 
 describe('Serialization tests', () => {
-  it('ser/deser string', () => {
+  it('ser/deser Utf-16 string', () => {
```
Comment (gregLibert) [inline packages/as-types/assembly/__tests__/serializer.spec.ts:23]: Are you sure it's UTF-16 encoded ? The file is not encoded in UTF8 ?

To be sure I would have used [String.fromCharCode()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/fromCharCode)

**Review by gregLibert — CHANGES_REQUESTED** [review]
:thinking:

File: packages/as-types/assembly/__tests__/serializer.spec.ts (hunk 2 above)
Comment (peterjah) [reply packages/as-types/assembly/__tests__/serializer.spec.ts:23]: the "🙂" emoji is [U+1F642](https://unicode.org/emoji/charts/full-emoji-list.html#1f642) unicode. which gives f0 9f 99 82 in utf8, so 4 bytes.
UTF-16 is variable length with code points encoded with one or two 16-bit code units
So the 🙂 emoji is a charactere encoded on 32 bits. and if we see its as an emoji, its mean its utf16 and your editor support it for display

File: packages/as-types/assembly/__tests__/serializer.spec.ts (hunk 1 above)
Comment (peterjah) [reply packages/as-types/assembly/__tests__/serializer.spec.ts:27]: Not sure to get your point. there is a test with emoji just before this one.

So in the end, whats the conclusion? you think my proposal its not working? should we add some tests ? or you think this is confusing?

File: packages/as-types/assembly/__tests__/serializer.spec.ts (hunk 2 above)
Comment (peterjah) [reply packages/as-types/assembly/__tests__/serializer.spec.ts:23]: but maybe i don't understand it properly....

File: packages/as-types/assembly/__tests__/serializer.spec.ts (hunk 1 above)
Comment (peterjah) [reply packages/as-types/assembly/__tests__/serializer.spec.ts:27]: I think my solution is working properly (unless someone prove me the opposite) and its saving some bytes in serialization.

File: packages/as-types/assembly/__tests__/serializer.spec.ts (hunk 2 above)
Comment (Thykof) [inline packages/as-types/assembly/__tests__/serializer.spec.ts:23]:
```suggestion
  it('ser/deser string with emoji', () => {
```

**Review by Thykof — APPROVED** [review]

**Review by gregLibert — APPROVED** [review]
:+1:

**Pull request merged**